### ISSUE TITLE
Variable Body File Name

### ISSUE THREAD

**Issue by kootstra** [opentable/wiremock-body-transformer#35]
On the wiremock-users mailing list archive I was looking for a solution for a variable BodyFile solution and was lead here with the expectation that this would be supported. I might be misinformed, which is why I'm creating this issue/question.

The aim of the rule is that I have several response file and I'd like to use the ID in the request to respond with the content of a file with the same name. Now, I am aware that I could create a rule for each case but having a variable bodyfile name would eliminate that. Does this project support this usecase?

**Comment by tranhungt** [comment]
The body transformer currently only supports parameterizing response body. It doesn't support dynamic file name, but it's possible, and requires what looks like a minor change.

I think you just need to pass in the `Map object` and do a regex match on filename inside the `getBodyResponse` method.

https://github.com/opentable/wiremock-body-transformer/blob/master/src/main/java/com/opentable/extension/BodyTransformer.java#L137

I don't have any time today, but if you or anyone would like to submit a PR, I'd be happy to review it.

**Comment by kootstra** [comment]
Thanks for the suggestion. I've been able to analyse it and I've made some modifications which I _think_ should work. However, I've been unable to compile them. That is likely due to my limited understanding of Eclipse and Java in general. So this is certainly something I can use some help with from you.

The idea is to use an additional Transformer Parameter tag body to hold a (partial) value received from the request that concatenates to the final file name:

```
{
    "request": {
        "method": "GET",
        "url": "/local-transform"
    },
    "response": {
        "status": 200,
        "transformers": ["body-transformer"],
		"TransformerParameter": [
			{
				"bodyFileName": {xPath request.body '/outer/inner/text()'}
			}
    }	]
}			
```

Line 137:
```
String responseBody = getResponseBody(responseDefinition, fileSource, parameters);
```

Line 185:
```
    private String getResponseBody(ResponseDefinition responseDefinition, FileSource fileSource, Parameters parameters) {
        String body, bodyFileName;
        if (responseDefinition.getBody() != null) {
            body = responseDefinition.getBody();
        } else {
        	
            if (parameters != null && parameters.containsKey("bodyFileName")) {
            	bodyFileName = parameters.getString("bodyFileName");
            } else {
            	bodyFileName = responseDefinition.getBodyFileName();
            }
        	
            BinaryFile binaryFile = fileSource.getBinaryFileNamed(bodyFileName);
            body = new String(binaryFile.readContents(), StandardCharsets.UTF_8);
        }
        return body;
    }

```

**Comment by kootstra** [comment]
The main issue I faced was that the project didn't function in Eclipse. When I moved to IntelliJ, the project worked out of the box. This made all the difference for me to run and debug the code changes I had made. In the end I opted to use the handlebar examples in the WireMock code and copied them to the plugin. The only supported functionality is to run the content of the BodyFileName through the handlebar processor and use it's result to actually load the file.
